File: PRG_410/AssignmentW4Night/AssignmentW4Night/Circle.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class Circle : IShape {


    public double Radius
    {
        set
        {
            if (value < 0)
                return;
            mRadius = value;
        }

        get
        {
            return mRadius;
        }
    }
    private double mRadius;

    public double Area() {
        // TODO implement here
        return Math.PI * Math.Pow(mRadius, 2);
        //return Math.PI * mRadius * mRadius;
    }

    /// <summary>
    /// 
    /// </summary>
    public void Foo()
    { }
    public double Perimeter() {
        // TODO implement here
        return 2 * mRadius * Math.PI;
    }

    /// <summary>
    /// @return
    /// </summary>
    public String Type() {
        // TODO implement here
        return "Circle";
    }

    /// <summary>
    /// @param radius
    /// </summary>
    public Circle(double radius) {
        mRadius = radius;
    }

    private Circle() {
        // TODO implement here
    }

}

File: PRG_410/WEEK_4/Assignment/code_Java/AssignmentWeek4/Circle.java

import java.util.*;

/**
 * 
 */
public class Circle implements IShape {

    /**
     * Default constructor
     */
    public Circle() {
    }

    /**
     * 
     */
    private double mRadius;

    /**
     * @return
     */
    public double Area() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @return
     */
    public double Perimeter() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @return
     */
    public String Type() {
        // TODO implement here
        return "";
    }

    /**
     * @param radius
     */
    public void Circle(double radius) {
        // TODO implement here
    }

    /**
     * 
     */
    private void Circle() {
        // TODO implement here
    }

    /**
     * @return
     */
    abstract double Area();

    /**
     * @return
     */
    public abstract double Perimeter();

    /**
     * @return
     */
    public abstract String Type();

}